python/lsst/ap/verify/ingestion.py

```diff
@@ -0,0 +1,480 @@
+    log.info('Images are now ingested in {0}'.format(repo))
+    metadata = ingestTask.getFullMetadata()
+    return metadata
+
+
+def _flatBiasIngest(repo, calibRepo, calibDataFiles):
+    """Ingest DECam flats and biases
+
+    Parameters
+    ----------
+    repo : `str`
+        The output repository location on disk where ingested raw images live.
+    calibRepo : `str`
+        The output repository location on disk where ingested calibration images live.
+    calibDataFiles : `list` of `str`
+        A list of the filenames of each flat and bias image file.
+
+    Returns
+    -------
+    metadata : `PropertySet` or `None`
+        Metadata from the `IngestCalibTask` (flats and biases) for use by ``ap_verify``
+    """
+    log = lsst.log.Log.getLogger('ap.pipe._flatBiasIngest')
+    log.info('Ingesting flats and biases...')
+    args = [repo, '--calib', calibRepo, '--mode', 'link', '--validity', '999']
+    args.extend(calibDataFiles)
+    argumentParser = IngestCalibsArgumentParser(name='ingestCalibs')
+    config = IngestCalibsConfig()
+    config.parse.retarget(ingestCalibs.DecamCalibsParseTask)
+    calibIngestTask = IngestCalibsTask(config=config, name='ingestCalibs')
+    parsedCmd = argumentParser.parse_args(config=config, args=args)
+    try:
+        calibIngestTask.run(parsedCmd)
+    except sqlite3.IntegrityError as detail:
+        log.error('sqlite3.IntegrityError: ', detail)
+        log.error('(sqlite3 doesn\'t think all the calibration files are unique)')
+        raise
+    else:
+        log.info('Success!')
+        log.info('Calibrations corresponding to {0} are now ingested in {1}'.format(repo, calibRepo))
+        metadata = calibIngestTask.getFullMetadata()
+    return metadata
+
+
+def _defectIngest(repo, calibRepo, defectFiles):
+    """Ingest DECam defect images
+
+    Parameters
+    ----------
+    repo : `str`
+        The output repository location on disk where ingested raw images live.
+    calibRepo : `str`
+        The output repository location on disk where ingested calibration images live.
+    defectFiles : `list` of `str`
+        A list of the filenames of each defect image file.
+        The first element in this list must be the name of a .tar.gz file
+        which contains all the compressed defect images.
+
+    Returns
+    -------
+    metadata : `PropertySet` or `None`
+        Metadata from the `IngestCalibTask` (defects) for use by ``ap_verify``
+
+    Notes
+    -----
+    This function assumes very particular things about defect ingestion:
+    - They must live in a .tar.gz file in the same location on disk as the other calibs
+    - They will be ingested using ingestCalibs.py run from the ``calibRepo`` directory
+    - They will be manually uncompressed and saved in :file:`calibRepo/defects/<tarballname>/`.
+    - They will be added to the calib registry, but not linked like the flats and biases
+    """
```

Comment:
Would be good to add a TODO in here referencing DM-5467 (i.e., it is known that DECam defect ingestion is awful and there is an outstanding ticket to make it less awful)